addons/payment/models/payment_transaction.py

```diff
@@ -1129,4 +1129,4 @@ def _get_last(self):
         :return: The last transaction of the recordset, sorted by id.
         :rtype: recordset of `payment.transaction`
         """
-        return self.filtered(lambda t: t.state != 'draft').sorted()[:1]
+        return self and self.filtered(lambda t: t.state != 'draft').sorted()[:1] or self
```

Comment:
Why did you need to change that? This seems wrong to me. The method would now not only return draft transactions, but also all transactions of the recordset (and not only the last one) when the `filtered` returns an empty recordset.

Reply:
In a call to `_get_last` the recordset self might be empty (partner is then set to false) and access rules will fail.

Reply:
still not good, I will change it again